From a817a1899909b9b210e0451339919253e49eeab6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dan=20Hor=C3=A1k?= <dan@danny.cz>
Date: Sun, 15 Jun 2014 11:08:37 +0200
Subject: [PATCH 1/3] detect version 1.x of libftdi library

The FDTI library changed version, module name and also soname, so add an option to detect it
when the 0.x version is not found. The 1.x API is compatible enough for libsigrok to build.
---
 configure.ac | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index cb4eeaf..ff93d8f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -370,7 +370,11 @@ PKG_CHECK_MODULES([libftdi], [libftdi >= 0.16],
 	[LIB_CFLAGS="$LIB_CFLAGS $libftdi_CFLAGS";
 	LIBS="$LIBS $libftdi_LIBS";
 	SR_PKGLIBS="$SR_PKGLIBS libftdi"],
-	[HW_ASIX_SIGMA="no"; HW_CHRONOVU_LA="no"; HW_IKALOGIC_SCANAPLUS="no"])
+	[PKG_CHECK_MODULES([libftdi1], [libftdi1 >= 1.1],
+		[LIB_CFLAGS="$LIB_CFLAGS $libftdi1_CFLAGS";
+		LIBS="$LIBS $libftdi1_LIBS";
+		SR_PKGLIBS="$SR_PKGLIBS libftdi1"],
+		[HW_ASIX_SIGMA="no"; HW_CHRONOVU_LA="no"; HW_IKALOGIC_SCANAPLUS="no"])])
 
 # The Check unit testing framework is optional. Disable if not found.
 PKG_CHECK_MODULES([check], [check >= 0.9.4],
@@ -599,7 +603,7 @@ echo "Detected libraries:"
 echo
 
 # Note: This only works for libs with pkg-config integration.
-for lib in "glib-2.0 >= 2.32.0" "libzip >= 0.10" "libserialport >= 0.1.0" "librevisa >= 0.0.20130812" "libusb-1.0 >= 1.0.16" "libftdi >= 0.16" "check >= 0.9.4"; do
+for lib in "glib-2.0 >= 2.32.0" "libzip >= 0.10" "libserialport >= 0.1.0" "librevisa >= 0.0.20130812" "libusb-1.0 >= 1.0.16" "libftdi >= 0.16" "libftdi1 >= 1.1" "check >= 0.9.4"; do
 	optional="OPTIONAL"
 	if test "x$lib" = "xglib-2.0 >= 2.32.0"; then optional="REQUIRED"; fi
 	if test "x$lib" = "xlibzip >= 0.10"; then optional="REQUIRED"; fi
-- 
1.9.3


From d9f2abd86839812ac06c61b97f4009a8e0276e7b Mon Sep 17 00:00:00 2001
From: Uwe Hermann <uwe@hermann-uwe.de>
Date: Mon, 16 Jun 2014 13:19:34 +0200
Subject: [PATCH 2/3] Lower libftdi1 requirement to 1.0 (builds fine too).

---
 configure.ac | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index ff93d8f..0212f05 100644
--- a/configure.ac
+++ b/configure.ac
@@ -370,7 +370,7 @@ PKG_CHECK_MODULES([libftdi], [libftdi >= 0.16],
 	[LIB_CFLAGS="$LIB_CFLAGS $libftdi_CFLAGS";
 	LIBS="$LIBS $libftdi_LIBS";
 	SR_PKGLIBS="$SR_PKGLIBS libftdi"],
-	[PKG_CHECK_MODULES([libftdi1], [libftdi1 >= 1.1],
+	[PKG_CHECK_MODULES([libftdi1], [libftdi1 >= 1.0],
 		[LIB_CFLAGS="$LIB_CFLAGS $libftdi1_CFLAGS";
 		LIBS="$LIBS $libftdi1_LIBS";
 		SR_PKGLIBS="$SR_PKGLIBS libftdi1"],
@@ -603,7 +603,7 @@ echo "Detected libraries:"
 echo
 
 # Note: This only works for libs with pkg-config integration.
-for lib in "glib-2.0 >= 2.32.0" "libzip >= 0.10" "libserialport >= 0.1.0" "librevisa >= 0.0.20130812" "libusb-1.0 >= 1.0.16" "libftdi >= 0.16" "libftdi1 >= 1.1" "check >= 0.9.4"; do
+for lib in "glib-2.0 >= 2.32.0" "libzip >= 0.10" "libserialport >= 0.1.0" "librevisa >= 0.0.20130812" "libusb-1.0 >= 1.0.16" "libftdi >= 0.16" "libftdi1 >= 1.0" "check >= 0.9.4"; do
 	optional="OPTIONAL"
 	if test "x$lib" = "xglib-2.0 >= 2.32.0"; then optional="REQUIRED"; fi
 	if test "x$lib" = "xlibzip >= 0.10"; then optional="REQUIRED"; fi
-- 
1.9.3


From d82281a6136dd2cf53cc058021842084b51d962c Mon Sep 17 00:00:00 2001
From: Uwe Hermann <uwe@hermann-uwe.de>
Date: Mon, 16 Jun 2014 13:20:56 +0200
Subject: [PATCH 3/3] README: Document that libftdi1 is also an option.

---
 README | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/README b/README
index 556ab3f..dc795f6 100644
--- a/README
+++ b/README
@@ -39,7 +39,7 @@ Requirements
  - libserialport >= 0.1.0 (optional, used by some drivers)
  - librevisa >= 0.0.20130812 (optional, used by some drivers)
  - libusb-1.0 >= 1.0.16 (optional, used by some drivers)
- - libftdi >= 0.16 (optional, used by some drivers)
+ - libftdi >= 0.16 or libftdi1 >= 1.0 (optional, used by some drivers)
  - check >= 0.9.4 (optional, only needed to run unit tests)
 
 
-- 
1.9.3

